src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -64,6 +64,14 @@ protected override string GetDatabaseFilePath(string workingFolderPath)
                 sqlStorage.Initialize(solution);
 
             }
+            catch (TypeInitializationException ex) when (ex.InnerException is DllNotFoundException || ex.InnerException is EntryPointNotFoundException)
+            {
+                // Probing for e_sqlite3 failed, so manually dispose the ownership lock as
+                // sqlStorage was not instantiated yet, due to an exception in the static constructor.
+                dbOwnershipLock.Dispose();
+                DisableStorage();
+                throw ex.InnerException;
```

Comment:
❓ Why not just `throw;` here?

Reply:
The exception thrown from here propagates into `ShouldDeleteDatabase`, via here:
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces.Desktop/Workspace/Storage/AbstractPersistentStorageService.cs,184

Maybe `throw;` is the better option here, and the check in `ShouldDeleteDatabase` should be done for TypeInitializationException and its inner exception.

Reply:
We're masking an exception and changing its stacktrace, so yes, I'll fix this.

Reply:
💭 It seems like the less risky change in the absence of a need to drop the outer exception.

Reply:
Fixed.